internal/utils/parse_cli_test.go

```diff
@@ -0,0 +1,59 @@
+package utils_test
+
+import (
+	"testing"
+
+	"github.com/rog-golang-buddies/internal/utils"
+)
+
+func TestParseCliArguments(t *testing.T) {
+
+	t.Run("Correctly gets commands", func(t *testing.T) {
+		commands, _ := utils.ParseCliArguments([]string{"test_command1", "test_command2"})
+
+		if len(commands) != 2 {
```

Comment:
These checks should be done in the ParseCliArguments function. The errors, if any, should be returned on there or nil.

Reply:
Sorry, I think I'm confused by your comment. How should the checks be done inside of the function since we don't know what elements of the array are commands or if they are flags

Reply:
I had thought about returning an error but I couldn't really justify it when thinking about how the function works. Imo there's no way for it to error since it's a pretty simple function

Reply:
I think of some edge cases that could happen:

- Setting a flag twice
- Adding "-", "--" or "---" without any name
- Specify commands after the flags: "gaa command1 command2 -flag1 aaa command3", is this allowed? error? optional?

Reply:
Ahh ok I understand. Will do! I'll update to add good error handling this week thank you very much for the feedback

Reply:
Addressed the comments discussed above. Also added additional unit tests to cover these cases